flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-paimon/src/main/java/org/apache/flink/cdc/connectors/paimon/sink/PaimonMetadataApplier.java

```diff
@@ -210,6 +209,21 @@ private List<SchemaChange> applyAddColumnEventWithPosition(AddColumnEvent event)
         try {
             List<SchemaChange> tableChangeList = new ArrayList<>();
             for (AddColumnEvent.ColumnWithPosition columnWithPosition : event.getAddedColumns()) {
+                // if default value express exists, we need to set the default value to the table
+                // option
+                Column column = columnWithPosition.getAddColumn();
+                Optional.ofNullable(column.getDefaultValueExpression())
+                        .ifPresent(
+                                value -> {
+                                    String key =
+                                            String.format(
+                                                    "%s.%s.%s",
+                                                    CoreOptions.FIELDS_PREFIX,
+                                                    column.getName(),
+                                                    CoreOptions.DEFAULT_VALUE_SUFFIX);
+                                    tableChangeList.add(SchemaChangeProvider.setOption(key, value));
```

Comment:
This can be removed as we have added the option in `SchemaChangeProvider#add` method, or we can remove the duplicated operation in `SchemaChangeProvider#add`.